src/compiler/lsp/protocol.cc

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2022 Toitware ApS.
```

Comment:
New file, but all of the code here has just been moved here (except some trivial forwarding functions).